app/src/main/java/org/mifos/mobilebanking/utils/ConfigurationPreference.java

```diff
@@ -0,0 +1,52 @@
+package org.mifos.mobilebanking.utils;
+
+import android.content.Context;
+import android.support.v7.preference.DialogPreference;
+import android.util.AttributeSet;
+
+import org.mifos.mobilebanking.R;
+import org.mifos.mobilebanking.api.BaseURL;
+import org.mifos.mobilebanking.api.SelfServiceInterceptor;
+
+/**
+ * Created by dilpreet on 11/03/18.
+ */
+
+public class ConfigurationPreference extends DialogPreference {
+
+    public ConfigurationPreference(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        init();
+    }
+
+    public ConfigurationPreference(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        init();
+    }
+
+    public ConfigurationPreference(Context context) {
+        super(context);
+        init();
+    }
+
+    private void init() {
+        setPersistent(false);
+    }
+
+    public String getBaseUrl() {
+        return getSharedPreferences().getString(getContext().getString(R.string.pref_base_url_key),
+                new BaseURL().getDefaultBaseUrl());
+    }
+
+    public void updateConfigurations(String baseUrl, String tenant) {
+        getSharedPreferences().edit()
```

Comment:
don't use getSharedPreferences, we already initialized the PreferencesHelper using a dagger, use that.